www/front_src/src/Navigation/Sidebar/Menu/CollapsableItems.tsx

```diff
@@ -236,26 +251,118 @@ const CollapsableItems = ({
     setCollapseScrollMaxWidth((window.innerWidth - rect.left) / 8);
   };
 
+  const fillItemsHoveredByDefault = (
+    rootItem: Page | null,
+    parentItem: Page | null,
+    childItem: Page | null,
+  ): void => {
+    rootItemHoveredByDefault.current = rootItem;
+    parentItemHoveredByDefault.current = parentItem;
+    childItemHoveredByDefault.current = childItem;
+  };
+
+  const searchItemsHoveredByDefault = (
+    childItem: Page | null,
+    parentItem: Page | null = null,
+  ): void => {
+    if (!isNil(childItem) && !childItem.is_react) {
+      const page = search?.match(/\d+/);
+
+      if (page) {
+        if (equals(page[0], childItem?.page)) {
+          if (getRootItemHoveredByDefault) {
+            fillItemsHoveredByDefault(
+              getRootItemHoveredByDefault(),
+              parentItem,
+              childItem,
+            );
+          }
+        }
+      }
+    } else if (!isNil(childItem) && equals(pathname, childItem?.url)) {
+      if (getRootItemHoveredByDefault) {
+        fillItemsHoveredByDefault(
+          getRootItemHoveredByDefault(),
+          parentItem,
+          childItem,
+        );
+      }
+    }
+
+    if (isArrayItem(childItem?.groups)) {
+      childItem?.groups?.forEach((itemGroup) => {
+        if (isArrayItem(itemGroup?.children)) {
+          itemGroup?.children?.forEach((i) =>
+            searchItemsHoveredByDefault(i, childItem),
+          );
+        }
+      });
+    }
+  };
+
+  const isItemHoveredByDefault = (item: Page): boolean => {
```

Comment:
Same remark as above, we should seek to reduce the complexity. I can see places where we could return the expressions directly for instance instead of a literal 'false'